std/utf.d

```diff
@@ -3106,17 +3106,31 @@ enum dchar replacementDchar = '\uFFFD';
  * Iterate a range of char, wchar, or dchars by code unit.
  *
  * The purpose is to bypass the special case decoding that
- * $(XREF array,front) does to character arrays.
+ * $(XREF array,front) does for character arrays. As a result, using ranges
+ * with `byCodeUnit` can be `nothrow` while $(XREF array,front) throws when
+ * it encounters invalid Unicode sequences.
+ *
+ * $(RED Warning:) This range does no encoding, so if the string you're passing
+ * to `byCodeUnit` contains things such as Unicode combining characters or
+ * graphemes that take up more than one code unit, then when iterating over
+ * this range will not encode those values into code points that would produce
+ * "correct" Unicode graphemes.
```

Comment:
This still seems more confusing than enlightening to me. A newbie is going to think: "I don't know what combining characters or graphemes are, so I'm probably not using them; therefore I'm fine."

I suggest to focus on "code unit" and leave other big words out:
    
    A code unit is a building block of the UTF encodings. Generally, an
    individual code unit does not represent what's perceived as a full
    character. Many characters are encoded with multiple code units. That
    means, an element of `byCodeUnit` often does not form a character on
    its own. Attempting to treat it as one will give nonsensical results.
    See `byGrapheme` for a range that emits full characters (graphemes).

(maybe a bit long)

I don't think we should mark this as a warning. It doesn't describe surprising behavior.

This overlaps with the paragraph in `See_Also`, of course. I'd just put a plain link to `byGrapheme` there, without additional explanation.

By the way, the transformation from code units to code points is decoding, not encoding. Encoding is the other way around: from code point to code units.